mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.feature
+
+import org.apache.spark.annotation.AlphaComponent
+import org.apache.spark.ml._
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.param.shared._
+import org.apache.spark.mllib.feature
+import org.apache.spark.mllib.linalg.{Vector, VectorUDT}
+import org.apache.spark.sql._
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{StructField, StructType}
+
+/**
+ * Params for [[MinMaxScaler]] and [[MinMaxScalerModel]].
+ */
+private[feature] trait MinMaxScalerParams extends Params with HasInputCol with HasOutputCol {
+
+  /**
+   * new minimum value after transformation, shared by all features
+   * Default: 0.0
+   * @group param
+   */
+  val lowerBound: DoubleParam = new DoubleParam(this, "lowerBound",
+    "lower bound of the expected feature range")
+
+  /**
+   * new maximum value after transformation, shared by all features
+   * Default: 1.0
+   * @group param
+   */
+  val upperBound: DoubleParam = new DoubleParam(this, "upperBound",
+    "upper bound of the expected feature range")
+}
+
+/**
+ * :: AlphaComponent ::
+ * Rescale original data values to a new range [lowerBound, upperBound] linearly using column
+ * summary statistics (minimum and maximum), which is also known as min-max normalization or
+ * Rescaling. The rescaled value for feature E is calculated as,
+ *
+ * Rescaled(e_i) = \frac{e_i - E_{min}}{E_{max} - E_{min}} * (upperBound - lowerBound) + lowerBound
+ */
+@AlphaComponent
+class MinMaxScaler extends Estimator[MinMaxScalerModel] with MinMaxScalerParams {
```

Comment:
This class should override Params.validateParams() and make sure lowerBound < upperBound.